sdk/formrecognizer/azure-ai-formrecognizer/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,129 @@
+       * [Invalid training dataset](#invalid-training-data-set)
+       * [Invalid SAS URL](#invalid-sas-url)
+    * [Generic Error](#generic-error)
+* [Unexpected time to build a custom model](#unexpected-time-to-build-a-custom-model)
+* [Enable HTTP request/response logging](#enable-http-requestresponse-logging)
+
+## Troubleshooting Errors & Exceptions
+### Handling HttpResponseException
+Form Recognizer service methods throw a [HttpResponseException](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/exception/HttpResponseException.java) or its subclass on failure.
+The HttpResponseException thrown by the Azure Form Recognizer client library includes detailed response error object
+that provides specific useful insights into what went wrong and include corrective actions to fix common issues.
+This error information can be found inside the message property of the HttpResponseException object.
+
+Here's the example of how to catch it with synchronous client
+
+```java readme-sample-handlingException
+try {
+    documentAnalysisClient.beginAnalyzeDocumentFromUrl("prebuilt-receipt", "invalidSourceUrl");
+} catch (HttpResponseException e) {
+    System.out.println(e.getMessage());
+    // Do something with the exception
+}
+```
+
+With async clients, you can catch and handle exceptions in the error callbacks:
+
+```java readme-sample-async-handlingException
+administrationAsyncClient.deleteModel("{modelId}")
+    .doOnSuccess(
+        ignored -> System.out.println("Success!"))
+    .doOnError(
+        error -> error instanceof ResourceNotFoundException,
+        error -> System.out.println("Exception: Delete could not be performed."));
+```
+
+### Build model error
+You may have seen a Build model error, mostly when you are trying to build your custom model.
+It looks like this:
+
+```text
+com.azure.core.exception.HttpResponseException: Invalid model created with model Id [modelId], errorCode: [2012], message: <specific-reason>"
+```
+
+#### Invalid training data set
+This error indicates that the provided data set does not match the training data requirements.
+Learn more about building a training data set, [here](https://aka.ms/customModelV3)
+The error looks like this:
+
+```json
+{
+"code": "InvalidRequest",
+"message": "Invalid request.",
+"innererror": {
+  "code": "ModelBuildError",
+  "message": "Could not build the model: OCR file 'Form_1.jpg.ocr.json' has an invalid schema."
+  }
+}
+```
+
+The exception looks like this:
+```text
+com.azure.core.exception.HttpResponseException: Invalid request, errorCode: [ModelBuildError], message: Could not build the model: OCR file 'Form_1.jpg.ocr.json' has an invalid schema"
+```
+
+#### Invalid SAS URL
+You may have seen this error like:
+```
+com.azure.core.exception.HttpResponseException: Invalid model created with model Id [modelId], errorCode: [2012], message: Unable to list blobs on the Azure blob storage account.
+```
```

Comment:
The other error sections put the example error below the description, this one should follow that pattern too?